src/SentryOne.UnitTestGenerator.Specs/Strategies/Class Generation/AbstractClassGeneration.feature

```diff
@@ -0,0 +1,155 @@
+﻿Feature: AbstractClassGeneration
+	I am checking the Abstract Class Generation strategy
+
+Scenario: Abstract Class Generation
+	Given I have a class defined as 
+		"""
+namespace TestNamespace {
+
+
+	public abstract class TestClass
+	{
+		protected TestClass(int intProperty, string value, Stream targetStream)
+		{
+
+		}
+ 
+		public abstract int SomeMethod();
+
+		public abstract void SomeVoidMethod();
+
+		protected abstract void SomeProtectedAbstractMethod();
+
+		public int IntProperty { get; set; }
+
+		protected int IntGetOnlyProperty { get; private set; }
+
+		protected string Value { get; set; }
+
+		protected string Value2 { get; private set; }
+
+		protected string Value3 { private get; set; }
+
+		protected abstract string AbstractValue { get; set; }
+
+		protected abstract string AbstractValue2 { get; }
+
+		protected abstract string AbstractValue3 { set; }
+
+		protected Stream TargetStream { get; }
+
+		protected string SomeMethod(int i, int j) { return string.Empty; }
+
+		protected abstract string SomeAbstractMethod(int i, int j);
+
+		public abstract string SomePublicAbstractMethod(int p, int r);
+
+		protected void SomeMethodVoid(int s, int t) {  }
+
+		protected internal void SomeMethodVoid2(int s, int t) {  }
+
+		protected internal abstract void SomeMethodVoid3(int s, int t);
+
+		private protected abstract void SomeMethodVoid4(int s, int t);
+	}
+}
+        """
+	And I set my test framework to 'NUnit3'
+	And I set my mock framework to 'NSubstitute'
```

Comment:
This is a good place to use Scenario Outlines (https://cucumber.io/docs/gherkin/reference/#scenario-outline) so that you can add more examples easily.